src/optimizer/query_rewrite.c

```diff
@@ -7628,6 +7628,38 @@ qo_rewrite_innerjoin (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *c
   return node;
 }
 
+/*
+ * qo_add_keylimit_clause () - Add limit clause to subquery exists
+ *   return: void
+ *   parser(in):
+ *   node(in): QUERY node
+ */
+void
+qo_add_limit_clause (PARSER_CONTEXT * parser, PT_NODE * node)
+{
+  bool has_instnum = false, has_orderbynum = false, has_groupbynum = false;
+
+  (void) parser_walk_tree (parser, node->info.query.q.select.where, pt_check_instnum_pre, NULL, pt_check_instnum_post,
+			   &has_instnum);
+  (void) parser_walk_tree (parser, node->info.query.orderby_for, pt_check_orderbynum_pre, NULL,
+			   pt_check_orderbynum_post, &has_orderbynum);
+  (void) parser_walk_tree (parser, node->info.query.q.select.having, pt_check_groupbynum_pre, NULL,
+			   pt_check_groupbynum_post, &has_groupbynum);
+
+  if (has_instnum || has_orderbynum || has_groupbynum)
+    {
+      return;			/* give up */
+    }
```

Comment:
`PT_UNION, PT_DIFFERENCE, PT_INTERSECTION`인 경우 때문에 `node->info.query.limit != NULL`에 대한 체크를 같이 해줘야 하네요.
```suggestion
  bool has_instnum = false, has_orderbynum = false, has_groupbynum = false;

  if (PT_IS_SELECT (node))
    {
      (void) parser_walk_tree (parser, node->info.query.q.select.where, pt_check_instnum_pre, NULL,
			       pt_check_instnum_post, &has_instnum);
      (void) parser_walk_tree (parser, node->info.query.orderby_for, pt_check_orderbynum_pre, NULL,
			       pt_check_orderbynum_post, &has_orderbynum);
      (void) parser_walk_tree (parser, node->info.query.q.select.having, pt_check_groupbynum_pre, NULL,
			       pt_check_groupbynum_post, &has_groupbynum);
    }

  if (node->info.query.limit != NULL || has_instnum || has_orderbynum || has_groupbynum)
    {
      return;			/* give up */
    }
```

Reply:
완료되었습니다. [df89349](https://github.com/CUBRID/cubrid/pull/6192/commits/df8934943f9b7f247cb79df640aec8adcad25405)

Reply:
확인했습니다. 감사합니다.